src/components/ui/rootHeader.tsx

```diff
@@ -0,0 +1,42 @@
+import Image from "next/image";
+import React from "react";
+
+import { CircleUser } from "lucide-react";
+import { RoutePath } from "@/lib/route-path";
+import { getLoggedInSession } from "@/lib/auth";
+import Link from "next/link";
+
+const RootHeader = async () => {
+  const session = await getLoggedInSession();
+  const accountType = session?.accountType;
+
+  return (
+    <nav className="sticky top-0 z-10 bg-white">
+      <div className=" flex items-center justify-between border-b shadow-lg">
+        <Link href="/">
+          <Image
+            src="/bark-bank-logo.svg"
+            alt="bark bank logo"
+            width={60}
+            height={60}
+            className="ml-4 md:ml-10"
+          />
+        </Link>
+
+        <div className="hidden gap-x-4 sm:flex sm:last:mr-10">
+          <Link href="/">Home</Link>
+          <Link href={RoutePath.ABOUT_US}>About Us</Link>
+          <Link href={RoutePath.BE_A_DONOR}>Be a Donor</Link>
+          <Link href={RoutePath.ARTICLES}>Articles</Link>
+          <Link href={RoutePath.FAQ}>FAQ</Link>
+          <Link href={RoutePath.INFO}>Info</Link>
+          <Link href={`/${accountType?.toLowerCase()}/dashboard`}>
+            <CircleUser />
+          </Link>
```

Comment:
I'm concerned that when we add a new account type, we might forget to check that this derived path makes sense. So it'll be better to add to RoutePath an ACCOUNT_DASHBOARD path.

It would be used like this `RoutePath.ACCOUNT_DASHBOARD(accountType)`.

There should be a test that fails the moment we introduce a new enum, but did not also update `RoutePath.ACCOUNT_DASHBOARD` to handle the case.

Reply:
Currently looking into this

```ts
```

Reply:
I saw this before.

I suspect it is the import AccountType from auth.ts. Causes backend stuff to be dragged into the client scope. Maybe move it into `auth-models.ts`? That way both client-side and backend code can import it without issue.

Reply:
Thanks for the suggestion! I believe your hunch is correct since `auth` is a server component that was being dragged into `bark-login-form` which is a client component.

Done! 209ecc4580a671adaddf1d84904b8197a4222d9a + ddd12d02d535243912b13e8b29818b19096d2c85

While performing this refactor I discovered something strange as shown in the attached image.

![Screenshot 2024-03-22 at 02 33 10](https://github.com/barkbank/barkbank-pawtal/assets/48792128/d02b53bd-a6c1-4b4f-aa31-b0a70e7f4ce2)

This is the reason why there are 2 commits, the first one didn't catch this.

Logging this shows `undefined` as it is technically valid JS, so the application works fine as well.

Could be a potential enhancement to the linter.

Reply:
Thanks for the refactor.

If I were to speculate—guess, hunch—it worked because the `bark-login-form.tsx` did not change and so a cached build of the page was used, which would work without issue since none of the account type definitions actually changed. A fresh rebuild would fail, however.

Could you help fix this other case of using `"USER"` as well?

https://github.com/barkbank/barkbank-pawtal/blob/ea6f95ebba58000146e1f1e02e1f0f17aee5f6e1/src/app/user/registration/_components/donor-form.tsx#L146-L153